web/src/navbar_alerts.ts

```diff
@@ -420,6 +464,35 @@ const time_zone_update_offer_banner = (): AlertBanner => {
     };
 };
 
+const upload_file_quota_warning_banner = (quota: number): AlertBanner => {
+    const is_limited_plan = realm.realm_plan_type === settings_config.realm_plan_types.limited.code;
+    return {
+        process: "upload_quota_warning",
+        intent: "warning",
+        label: $t(
+            {
+                defaultMessage:
+                    "Your organization is using {percentage_upload_space}% of your {upload_quota}file storage quota.{upgrade_message}",
+            },
+            {
+                percentage_upload_space,
+                upload_quota: `${bytes_to_size(mib_to_bytes(quota), true)} `,
```

Comment:
It would probably be cleaner to encapsulate the calculations for `percentage_upload_space` and `upload_quota` within `attachments_ui.ts` as getter functions, and simply access their values here for the sake of clearer separation of concerns and keeping the logic modular.